app/design/frontend/Magebit/Learning/Magento_Catalog/templates/product/view/addtocart.phtml

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * @copyright Copyright (c) 2024 Magebit (https://magebit.com)
+ * @author    Magebit <info@magebit.com>
+ * @license   GNU General Public License ("GPL") v3.0
+ */
+/** @var Magento\Catalog\Block\Product\View $block */
+/** @var \Magento\Catalog\Model\Product $product */
+/** @var \Magento\CatalogInventory\Api\StockRegistryInterface $stockRegistry */
+$product = $block->getProduct();
+$objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+$stockRegistry = $objectManager->get(\Magento\CatalogInventory\Api\StockRegistryInterface::class);
```

Comment:
If you want to use some other classes that are not in this particular block, you need to create your own block. Possibly extending the block you are using right now, so you inherit all of the original block methods.
Then add methods you need like getStockItem etc.